repos/system_upgrade/el7toel8/actors/nischeck/libraries/nischeck.py

```diff
@@ -0,0 +1,150 @@
+import ipaddress
+import re
+
+from leapp import reporting
+
+re_server = r"\s*domain\s+\S+\s+server\s+(\S+)\s*"
+re_ypserver = r"\s*ypserver\s+(\S+)\s*"
+
+
+def scan():
+    """
+    Check configuration of /etc/yp.conf and /etc/nsswitch.conf. Address of NIS server cannot be
+    specified by host name if NIS is used to resolve host names.
+    """
+
+    hostnames = hostnames_in_yp_conf()
+    data, _ = scan_nsswitch()
+
+    if "hosts" in data:
+        if "nis" in data["hosts"] and hostnames:
+            report_error(hostnames)
+
+
+def report_error(hostnames):
+    reporting.create_report([
+        reporting.Title('Unsupported NIS configuration found'),
+        reporting.Summary("NIS may be used for domain name resolution only if NIS "
+                          "server is specified by IP. NIS servers specified by "
+                          "host name: {}".format(", ".join(hostnames))),
+        reporting.Severity(reporting.Severity.MEDIUM)
+    ])
+
+
+def hostnames_in_yp_conf():
+    """
+    Scan /etc/yp.conf, return if file contains domain name
+    """
+
+    try:
+        with open("/etc/yp.conf") as f:
+            lines = f.readlines()
```

Comment:
@fila43 actors in the checks phase cannot access anything on the system. expected solution is
* create the scanner actor that scan the configution files and provide the data
* update the check actor to consume the data and provide report when needed

Reply:
@fila43 but let's sync about that. I see there have been already couple of reviews and unfortunately nobody checked that - we are sorry about that. so let's rather sync about this PR in person first to discuss the PR first. We do not want to create for people additional burden based on our mistakes.